app/models/batch_connection.rb

```diff
@@ -3,4 +3,17 @@
 class BatchConnection < ApplicationRecord
   belongs_to :batch_process
   belongs_to :connectable, polymorphic: true
+
+  def child_object_count
+    return unless connectable_type == "ParentObject"
+    ParentObject.find(connectable_id)&.child_object_count
+  rescue ActiveRecord::RecordNotFound
+    nil
+  end
+
+  def status
+    ParentObject.find(connectable_id).status_for_batch_process(batch_process_id)
```

Comment:
you shouldn't ever need to do PO.find here.

`connection&.status_for_batch_process(batch_process_id) || "deleted"` makes more sense to me, would be a lot more efficient and might work in all cases